io.openems.edge.evcs.api/src/io/openems/edge/evcs/v2/api/Evcs.java

```diff
@@ -0,0 +1,846 @@
+	 * 
+	 * <ul>
+	 * <li>Interface: Evcs
+	 * <li>Readable
+	 * <li>Type: Integer
+	 * <li>Unit: Milliampere
+	 * </ul>
+	 */
+	CURRENT_L3(Doc.of(OpenemsType.INTEGER) //
+		.unit(Unit.MILLIAMPERE) //
+		.accessMode(AccessMode.READ_ONLY) //
+		.persistencePriority(PersistencePriority.HIGH)),
+
+	/**
+	 * Count of phases, the EV is charging with.
+	 *
+	 * <p>
+	 * This value is derived from the charging station or calculated during the
+	 * charging. When this value is set, the minimum and maximum limits are set at
+	 * the same time if the EVCS is a {@link ManagedEvcs}.
+	 * 
+	 * <p>
+	 * phases must be set to 3 if undefined or unclear.
+	 * 
+	 * <ul>
+	 * <li>Interface: Evcs
+	 * <li>Readable
+	 * <li>Type: Phases @see @link {@link Phases}
+	 * </ul>
+	 */
+	PHASES(Doc.of(Phases.values()) //
+		.debounce(5) //
+		.accessMode(AccessMode.READ_ONLY) //
+		.persistencePriority(PersistencePriority.HIGH)), //
+
+	/**
+	 * Phase Rotation.
+	 * 
+	 * <p>
+	 * In charge parks the phases of chargepoints are rotated when installed. This
+	 * reduces phase shifting load when the majority of vehicles charge with only
+	 * 2phase or less. Channel provides information on the phase rotation of this
+	 * chargepoint.
+	 *
+	 * <ul>
+	 * <li>Interface: Evcs
+	 * <li>Readable
+	 * <li>Type: PhaseRotation @see {@link PhaseRotation}
+	 * </ul>
+	 */
+	PHASE_ROTATION(Doc.of(PhaseRotation.values()) //
+		.accessMode(AccessMode.READ_ONLY) //
+		.persistencePriority(PersistencePriority.HIGH)), //
+
+	/**
+	 * Energy that was charged during the current or last Session.
+	 *
+	 * <ul>
+	 * <li>Interface: Evcs
+	 * <li>Readable
+	 * <li>Type: Integer
+	 * <li>Unit: Wh
+	 * </ul>
+	 */
+	ENERGY_SESSION(Doc.of(OpenemsType.INTEGER) //
+		.unit(Unit.WATT_HOURS) //
+		.accessMode(AccessMode.READ_ONLY) //
+		.persistencePriority(PersistencePriority.HIGH)), //
+
+	/**
+	 * Active Consumption Energy.
+	 *
+	 * <ul>
+	 * <li>Interface: Evcs
+	 * <li>Type: Integer
+	 * <li>Unit: Wh
+	 * </ul>
+	 */
+	ACTIVE_CONSUMPTION_ENERGY(Doc.of(OpenemsType.LONG) //
+		.unit(Unit.WATT_HOURS) //
+		.accessMode(AccessMode.READ_ONLY) //
+		.persistencePriority(PersistencePriority.HIGH)), //
+
+	/**
+	 * Failed state channel for a failed communication to the EVCS.
+	 *
+	 * <ul>
+	 * <li>Interface: Evcs
+	 * <li>Readable
+	 * <li>Level: FAULT
+	 * </ul>
+	 */
+	COMMUNICATION_FAILED(Doc.of(Level.FAULT) //
+		.accessMode(AccessMode.READ_ONLY) //
+		.persistencePriority(PersistencePriority.HIGH)); //
+
+	private final Doc doc;
+
+	private ChannelId(Doc doc) {
+	    this.doc = doc;
+	}
+
+	@Override
+	public Doc doc() {
+	    return this.doc;
+	}
+    }
+
+    /**
+     * Initializes Channel listeners to sum current for L1 + L2 + L3 and add the
+     * value to the current channel.
+     *
+     * @param evcs the {@link Evcs}
+     */
+    public static void initializeCurrentSumChannels(Evcs evcs) {
+	final Consumer<Value<Integer>> currentSum = ignore -> {
+	    evcs._setCurrent(TypeUtils.sum(//
```

Comment:
that should not be the sum. if you have a three phase vehicle and the controller will set each phase to "6A", than the sum would be 18A. But internal the charger will have only "6A". that would be a bit confusing, wouldn't it?

Reply:
Good point. Replaced the method with an initializer to set current = max (currentL1, currentL2, currentL3). Why max? Chargepoints should have same current on all used phases. But with max() we are always on the safe side.